client/components/idealPayments/html/README.md

```diff
@@ -0,0 +1,162 @@
+
+## 🏗️ Architecture Overview
+
+This sample demonstrates a complete iDEAL integration flow:
+
+1. Initialize PayPal Web SDK with the iDEAL component
+2. Check eligibility for iDEAL payment method
+3. Create iDEAL payment session with required payment fields
+4. Validate customer information before initiating payment
+5. Process payment through iDEAL popup flow with bank selection
+6. Handle payment approval, cancellation, and errors
+
+## 📋 Prerequisites
+
+Before running this demo, you'll need to set up accounts and configure your development environment.
+
+### 1. PayPal Developer Account Setup
+
+1. **PayPal Developer Account**
+   - Visit [developer.paypal.com](https://developer.paypal.com)
+   - Sign up for a developer account or log in with existing credentials
+   - Navigate to the **Apps & Credentials** section in your dashboard
+
+2. **Create a PayPal Application** (or configure the default application)
+   - Click **Create App**
+   - Name your app
+   - Select **Merchant** under **Type**
+   - Choose the **Sandbox** account for testing
+   - Click **Create App** at the bottom of the modal
+   - Note your **Client ID** and **Secret key** under **API credentials** for later configuration of the `.env` file
+
+3. **Enable iDEAL Payment**
+   - Visit [sandbox.paypal.com](https://www.sandbox.paypal.com)
+   - Log in with your **Sandbox** merchant account credentials
+   - Navigate to **Account Settings** by clicking the profile icon in the top right corner
+   - Select **Payment methods** from the left sidebar
+   - Find **iDEAL** in the payment methods and enable it
+
+### 2. System Requirements
+
+- Node.js version 20 or higher
+- Server running on port 8080 (see `server/node/` directory)
+
+## Features
+
+- iDEAL one-time payment integration
+- Full name field validation
+- Popup payment flow
+- Eligibility checking for iDEAL
+- Error handling and user feedback
+
+## 🚀 Running the Demo
+
+### Server Setup
+
+1. **Navigate to the server directory:**
+
+   ```bash
+   cd server/node
+   ```
+
+2. **Install dependencies:**
+
+   ```bash
+   npm install
+   ```
+
+3. **Configure environment variables:**
+   Create a `.env` file in the root directory using your client credentials from the previous Create Application step:
+
+   ```env
+   PAYPAL_SANDBOX_CLIENT_ID=your_paypal_sandbox_client_id
+   PAYPAL_SANDBOX_CLIENT_SECRET=your_paypal_sandbox_client_secret
+   ```
+
+4. **Start the server:**
+   ```bash
+   npm start
+   ```
+   The server will run on `http://localhost:8080`
+
+### Client Setup
+
+1. **Navigate to the iDEAL demo directory:**
+
+   ```bash
+   cd client/components/idealPayments/html
+   ```
+
+2. **Install dependencies:**
+
+   ```bash
+   npm install
+   ```
+
+3. **Start the development server:**
+   ```bash
+   npm start
+   ```
+   The demo will be available at `http://localhost:3000`
+
+## How It Works
+
+### Client-Side Flow
```

Comment:
Would it be helpful to mention in this section where the server-side endpoints are used?